python/whylogs/core/preprocessing.py

```diff
@@ -250,6 +256,46 @@ def _process_scalar_value(value: Any) -> "PreprocessedColumn":
 
         return result
 
+    @staticmethod
+    def _process_homogeneous_column(series: pd.Series) -> "PreprocessedColumn":
+        """
+        Column must be of homogeneous type. NaN, None, other missing data not allowed.
+        """
+
+        result = PreprocessedColumn()
+        result.original = series
+        result.len = len(series)
+        if series.empty:
+            return result
+
+        if pdc.is_numeric_dtype(series.dtype) and not pdc.is_bool_dtype(series.dtype):
+            if pdc.is_float_dtype(series.dtype):
+                result.numpy.floats = series.astype(float)
+                return result
+            else:
+                result.numpy.ints = series.astype(int)
+                return result
+
+        value = series[0]
```

Comment:
comment here to draw attention that this is the unique aspect of the homogenous processing